solr/core/src/java/org/apache/solr/handler/designer/DefaultSampleDocumentsLoader.java

```diff
@@ -65,14 +64,11 @@ public class DefaultSampleDocumentsLoader implements SampleDocumentsLoader {
   private static final int MAX_STREAM_SIZE = (5 * 1024 * 1024);
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
-  public static byte[] streamAsBytes(final InputStream in) throws IOException {
-    ByteArrayOutputStream baos = new ByteArrayOutputStream();
-    byte[] buf = new byte[1024];
-    int r;
+  /** Reads input completely into a byte array. Closes the stream. */
```

Comment:
This feels super trappy to have this method close the stream and Utils.toByteArray leave it open, but not something we can fix here.

Reply:
Yes; it's unusual for a method to accept an InputStream and also close it.  I at least documented it.  And I reduced its scope from public.  I was tempted to rename the method to something like streamAsBytesAndClose so that the closing is super apparent at the caller.  Can still do.

Reply:
we have the `andClose` pattern in other places...